sqlmesh/core/linter/rule.py

```diff
@@ -50,6 +50,19 @@ def __repr__(self) -> str:
         return f"{self.rule.name}: {self.violation_msg}"
 
 
+class RuleViolationWithModelAndType(RuleViolation):
```

Comment:
Should we just shove the new logic in `RuleViolation`? Model and violation type seems useful in general?

Reply:
To me, here you kind of need two classes: 
1. `RuleViolation` is the neat thing that a Rule emits
2. `RuleViolationWithModelAndType` is the type that comes from the rule where we have augmented it with severity and the rule that went in.